src/renderer/hooks/useMapStylesQuery.js

```diff
@@ -42,7 +42,9 @@ export const useMapStylesQuery = (enabled = true) => {
         refetch: mapStylesQueryResult.refetch
       }
     : {
-        data: legacyStyleQueryResult.data,
+        data: !legacyStyleQueryResult.data
+          ? [defaultMapStyle]
+          : legacyStyleQueryResult.data,
```

Comment:
this encapsulates the undefined check so the consuming components do not have to do that